File: problems/chapter12/miiitomi/002.cpp
// AtCoder ABC121   C - Energy Drink Collector
// https://atcoder.jp/contests/abc121/tasks/abc121_c

#include <bits/stdc++.h>
using namespace std;

int main() {
    int N, M;
    cin >> N >> M;
    vector<pair<int, int>> V(N);
    for (int i = 0; i < N; i++) {
        int a, b;
        cin >> a >> b;
        V[i] = make_pair(a, b);
    }
    sort(V.begin(), V.end());
    int64_t ans = 0LL;
    for (int i = 0; i < N; i++) {
        if (M > V[i].second) {
            M -= V[i].second;
            ans += (int64_t)V[i].first * V[i].second;
        } else {
            ans += (int64_t)V[i].first * M;
            break;
        }
    }
    cout << ans << endl;
}

File: problems/chapter12/mshibatatt/002.py
# https://atcoder.jp/contests/abc121/tasks/abc121_c

def main():
    N, M = map(int, input().split())
    stores = []
    for _ in range(N):
        stores.append(list(map(int, input().split())))

    stores.sort()
    output = 0
    counter = 0
    i = 0
    while counter < M:
        output += stores[i][0] * min(stores[i][1], M - counter)
        counter += stores[i][1]
        i += 1

    print(output)

if __name__=='__main__':
    main()